tutorials/crx-extensions/src/main/java/WebStoreDownload.java

```diff
@@ -0,0 +1,82 @@
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ *  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ *  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ *  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import static com.teamdev.jxbrowser.engine.RenderingMode.HARDWARE_ACCELERATED;
+import static javax.swing.WindowConstants.DISPOSE_ON_CLOSE;
+
+import com.teamdev.jxbrowser.engine.Engine;
+import com.teamdev.jxbrowser.extensions.callback.InstallExtensionCallback;
+import com.teamdev.jxbrowser.view.swing.BrowserView;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+
+/**
+ * Use this application to download a CRX file from Chrome Web Store.
+ *
+ * <p>To download a CRX file, launch the application, then find and install the
+ * extension. Once the CRX files is downloaded, the application will close.
+ */
+public final class WebStoreDownload {
+
+    @SuppressWarnings("resource")
+    // The engine and browser will be closed when the Swing window is closed.
+    public static void main(String[] args) {
+        var engine = Engine.newInstance(HARDWARE_ACCELERATED);
+        var browser = engine.newBrowser();
+        browser.navigation().loadUrl(
+                "https://chromewebstore.google.com/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi");
+
+        SwingUtilities.invokeLater(() -> {
+            var view = BrowserView.newInstance(browser);
+            var frame = new JFrame("Chrome Web Store");
+            frame.add(view);
+            frame.setSize(1280, 900);
+            frame.setLocationRelativeTo(null);
+            frame.addWindowListener(new WindowAdapter() {
+                @Override
+                public void windowClosing(WindowEvent e) {
+                    engine.close();
+                }
+            });
+            frame.setDefaultCloseOperation(DISPOSE_ON_CLOSE);
+            frame.setVisible(true);
+
+            var extensions = browser.profile().extensions();
+            extensions.set(InstallExtensionCallback.class, (params, tell) -> {
+                var name = params.extensionName();
+                var sourceCrxFilePath = Paths.get(params.extensionCrxFile());
+                var targetCrxFilePath = Paths.get(name + ".crx");
+                try {
+                    Files.copy(sourceCrxFilePath, targetCrxFilePath);
+                    engine.close();
```

Comment:
It's better show the directory in Finder/Explorer where the downloaded CRX file is located now instead of closing the app. If we just close the app, it will look like crash.